src/main.c

```diff
@@ -80,6 +108,71 @@ static void load_embedded_startup_dmp(void)
     }
 }
 
+static void load_embedded_modules_dmp(void)
+{
+    void* modules_dmp_start = &__modules_dmp_start;
+    void* modules_dmp_end   = &__modules_dmp_end;
+    size_t modules_dmp_size = (size_t)((uintptr_t)modules_dmp_end - (uintptr_t)modules_dmp_start);
+    if(modules_dmp_size > 0)
+    {
+        DMOD_LOG_INFO("Loading modules.dmp from ROM: addr=0x%X, size=%u bytes\n", 
+                      (uintptr_t)modules_dmp_start, (unsigned int)modules_dmp_size);
+        
+        uint32_t package_index;
+        if(!Dmod_AddPackageBuffer(modules_dmp_start, modules_dmp_size, &package_index))
+        {
+            DMOD_LOG_ERROR("Failed to load modules.dmp package\n");
+        }
+        else
+        {
+            DMOD_LOG_INFO("Modules package loaded successfully (index=%u)\n", package_index);
+        }
+    }
+    else
+    {
+        DMOD_LOG_INFO("No modules.dmp embedded in ROM\n");
+    }
+}
+
+static void mount_embedded_romfs(void)
+{
+    void* romfs_start = &__romfs_start;
+    void* romfs_end   = &__romfs_end;
+    size_t romfs_size = (size_t)((uintptr_t)romfs_end - (uintptr_t)romfs_start);
+    if(romfs_size > 0)
+    {
+        DMOD_LOG_INFO("Mounting ROMFS from ROM: addr=0x%X, size=%u bytes\n", 
+                      (uintptr_t)romfs_start, (unsigned int)romfs_size);
+        
+        // Load dmffs module
+        if(!Dmod_LoadModuleByName("dmffs"))
+        {
+            DMOD_LOG_ERROR("Failed to load dmffs module\n");
+            return;
+        }
+        
+        // Mount the ROMFS at /romfs/
+        // The dmffs module expects parameters: <binary_data_addr> <mount_point>
+        char addr_str[32];
+        uint_to_hex_string((uint32_t)(uintptr_t)romfs_start, addr_str, sizeof(addr_str));
+        char* argv[] = { "dmffs", addr_str, "/romfs/" };
+        int argc = 3;
+        
+        if(Dmod_RunModule("dmffs", argc, argv) != 0)
```

Comment:
Przecież to kompletnie źle jest... Widać, że nawet tego nie testowałeś... Masz po to emulator skonfigurowany, żebys mógł testować. Do dupy taki kod jest